### PR TITLE
Missing label in ie

### PR DESCRIPTION
**Pull request by uplne**
We have missing 'Advertisement' label in IE<11.

DFP sometimes returns two iframes and one of them will never go to readyState. That with combination of some, probably accidental, changes to return value meant that condition to add label where not met. And this is specific to IE.

In this PR I've also added test to make sure that we don't add label only in the case when we get 'gu-style' response.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/dfp-api.js [hunk 1]
```diff
@@ -645,32 +643,38 @@ define([
...
+                        ) {
+                            bean.off(updatedIFrame, 'readystatechange');
+                            resolve(breakoutIFrame(updatedIFrame, $slot));
+                        }
+                    });
+                } else {
+                    resolve(breakoutIFrame(iFrame, $slot));
+                }
+            }).then(function (items) {
+                return new Promise.resolve(find(items, function (item) {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/dfp-api.js]: You don't need to use `Promise.resolve` in a `then` callback: `return` is how you implicitly resolve a Promise, while `throw` is how you reject it. This is in stark contrast with how it works inside a `new Promise(fn)`, but the constructor was never meant to be used directly.

File: static/src/javascripts/projects/common/modules/commercial/dfp-api.js [hunk 2]
```diff
@@ -645,32 +643,38 @@ define([
...
+                var iFrame = find($('iframe', $slot), function (iframe) { return iframe.id.match('__hidden__') === null; });
+
+                // No iFrame, no work to do
+                if (typeof iFrame === 'undefined') {
+                    reject();
+                }
+                // IE needs the iFrame to have loaded before we can interact with it
+                else if (iFrame.readyState && iFrame.readyState !== 'complete') {
+                    bean.on(iFrame, 'readystatechange', function (e) {
+                        var updatedIFrame = e.srcElement;
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/dfp-api.js]: Just to make sure, would this make sense: `updateIFrame = e.srcElement || e.target`

**Comment by uplne** [comment]
@regiskuckaertz this bad boy is ready to go out

**Comment by regiskuckaertz** [comment]
let's make this beast roar :+1:

**Pull request merged**